x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/Driver.java

```diff
@@ -394,18 +408,23 @@ void onComplete(ActionListener<Void> listener) {
         });
     }
 
-    private static SubscribableListener<Void> oneOf(List<SubscribableListener<Void>> futures) {
-        if (futures.isEmpty()) {
+    private static IsBlockedResult oneOf(List<IsBlockedResult> results) {
+        if (results.isEmpty()) {
             return Operator.NOT_BLOCKED;
         }
-        if (futures.size() == 1) {
-            return futures.get(0);
+        if (results.size() == 1) {
+            return results.get(0);
         }
         SubscribableListener<Void> oneOf = new SubscribableListener<>();
-        for (SubscribableListener<Void> fut : futures) {
-            fut.addListener(oneOf);
+        StringBuilder reason = new StringBuilder();
+        for (IsBlockedResult r : results) {
+            r.listener().addListener(oneOf);
+            if (reason.isEmpty() == false) {
+                reason.append(" OR ");
+            }
+            reason.append(r.reason());
         }
-        return oneOf;
+        return new IsBlockedResult(oneOf, reason.toString());
```

Comment:
This string is the key in a hash table but I think in practice we won't have that many of them so I think it's ok. Unless we make a bunch more async operators. Then maybe we'd prefer just taking the first one.